python/python_helper_sympy.py

```diff
@@ -221,64 +273,70 @@ def evaluate(expr, locals_for_eval={}):
 
     # Convert AST to code and evaluate it with no global expressions and with
     # a whitelist of local expressions
-    locals = {}
-    for key in locals_for_eval:
-        locals = {**locals, **locals_for_eval[key]}
+    locals = dict(chain.from_iterable(
+        cast(SympyMapT, local_expressions).items() for local_expressions in locals_for_eval.values()
+    ))
```

Comment:
why this change? I think the original was more readable.

Reply:
The original is wasteful (recreates the dictionary several times). There's a more readable way to write this though, so I'll use that.